### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by molly-moen**
# Description and Motivation 
Update readme to be in line with template and include more information

# Change type 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,84 +2,149 @@
...
 
-The service uses the information manage by PCS Telemetry microservice
-to set the stream processing logic, in order to implement monitoring
+The service uses the information managed by 
+the [Telemetry Microservice][telemetry-microservice-url]
+to set the stream processing logic, by implementing monitoring
 rules applied to the telemetry generated by IoT Devices.
 
-ASA jobs use a reference table, in order to support device grouping,
+ASA jobs use a reference table in order to support device grouping,
```
Comment (dluc) [inline README.md]: there are actually two reference data files (one is a CSV table, the other a JSON file)

Comment (molly-moen) [reply README.md]: Aren't they both json? One for rules, one for device groups? I'll update it

File: README.md [hunk 2]
```diff
@@ -100,33 +169,43 @@ the IDE, there are several ways to manage environment variables:
...
-
-Feedback
-==========
+# Troubleshooting
 
-Please enter issues, bugs, or suggestions as GitHub Issues here: 
-https://github.com/Azure/asa-manager-dotnet/issues
+It is possible Azure Stream Analytics may crash. If this happens (rules stop evaluating), 
+restart ASA in the portal with the following steps:
+1. Navigate to your ASA instance on portal.azure.com. If you deployed an entire
```
Comment (zhang-hua) [inline README.md]: use linkage for azure portal?

File: README.md [hunk 3]
```diff
@@ -2,84 +2,153 @@
...
+* **DeviceGroupsAgent.csproj** - C# assembly containing business logic
+  for querying device group to device id mapping and writing that mapping
+  to Azure Blob Storage
+* **DeviceGroupsAgent.Test.csproj** - Unit tests for device groups agent functionality
+* **TelemetryRulesAgent.csproj** - C# assembly containing business logic
+  for checking whether telemetry rules have changed, and exporting rules to ASA
+  when they change
+* **TelemetryRulesAgent.Test.csproj** -Unit tests for telemetry rules agent functionality
+* **Solution/scripts** - Contains build scripts, docker container creation scripts, 
+and scripts for running the microservice from the command line
```
Comment (zhang-hua) [inline README.md]: ASA query, UDF functions and device group and rules samples, and shell script to convert query and UDF functions into an ARM parameter object.

Comment (zhang-hua) [reply README.md]: maybe just append it to the end of above text.

File: README.md [hunk 4]
```diff
@@ -2,84 +2,156 @@
...
+* **DeviceGroupsAgent.Test.csproj** - Unit tests for device groups agent functionality
+* **TelemetryRulesAgent.csproj** - C# assembly containing business logic
+  for checking whether telemetry rules have changed, and exporting rules to ASA
+  when they change
+* **TelemetryRulesAgent.Test.csproj** -Unit tests for telemetry rules agent functionality
+* **Solution/scripts** - Contains build scripts, docker container creation scripts, 
+and scripts for running the microservice from the command line.
+* **Solution/scripts/asa** - Contains ASA query, UDF functions and device
+  group and rules samples, and shell script to convert query and UDF functions
+  into an ARM parameter object.
```
Comment (ppathan) [inline README.md]: nit: May be this is better: `Contains ASA query, UDF functions, device group, rules samples and shell script to convert query and UDF functions into an ARM parameter object`

File: README.md [hunk 5]
```diff
@@ -2,84 +2,155 @@
 [![Issues][issues-badge]][issues-url]
 [![Gitter][gitter-badge]][gitter-url]
 
-Azure Stream Analytics Manager overview
-=======================================
+# Azure Stream Analytics Manager Overview
 
-This service allows to manage the Azure Stream Analytics (ASA) jobs,
-to set the configuration, start, stop, and monitor their status.
+This service allows the management the Azure Stream Analytics (ASA) jobs,
```
Comment (dluc) [inline README.md]: missing "of"

File: README.md [hunk 6]
```diff
@@ -2,84 +2,155 @@
...
+including setting the configuration and starting, stopping, and 
+monitoring their statuses.
 
-The service uses the information manage by PCS Telemetry microservice
-to set the stream processing logic, in order to implement monitoring
+The ASA job is supported by two 
+[reference data](https://docs.microsoft.com/en-us/azure/stream-analytics/stream-analytics-use-reference-data)
+sets--one that defines rules and one that defines device groups.
+Both are stored in json format in Azure Blob Storage.
```
Comment (dluc) [inline README.md]: isn't one CSV and one JSON?

File: README.md [hunk 7]
```diff
@@ -2,84 +2,155 @@
...
+
+The rules reference data is generated by the information managed by 
+the [Telemetry Microservice][telemetry-microservice-url].
+This microservice sets the stream processing logic by implementing monitoring
```
Comment (dluc) [inline README.md]: add a blank line before this statement

Comment (dluc) [reply README.md]: I would rephrase, I find it hard to understand what is what in these sentences. For instance you can use "transform telemetry rules into stream processing logic" or something similar

File: README.md [hunk 8]
```diff
@@ -2,84 +2,155 @@
...
 rules applied to the telemetry generated by IoT Devices.
 
-ASA jobs use a reference table, in order to support device grouping,
-i.e. to define monitoring rules on a group of devices. The device
-groups are managed by PCS Configuration, using Azure IoT Hub device
-twin queries.
+The device groups reference data is used to
+define monitoring rules on a group of devices.
```
Comment (dluc) [inline README.md]: seems incorrect, or ambiguous.  This ref data is used to identify which group of rules to apply to an incoming telemetry message

File: README.md [hunk 9]
```diff
@@ -2,84 +2,155 @@
...
-* [Azure Stream Analytics][asa-url]
-* Azure service account, used to access and manage ASA configuration 
-* [PCS Telemetry microservice][telemetry-microservice-url] used to download
-  the remote monitoring rules, which are transformed into ASA Jobs logic
-* [PCS Configuration microservice][configuration-microservice-url] used
-  to download the list of device groups
-* Configuration settings to connect to the dependencies. These settings
-  are stored in environment variables, which are referenced by the service
-  configuration.
+* Managa ASA jobs
```
Comment (dluc) [inline README.md]: typo

File: README.md [hunk 10]
```diff
@@ -2,84 +2,155 @@
...
-```
-cd scripts
-cd docker
-docker-compose up
-```
+## Project Structure
 
-The Docker compose configuration requires the StorageAdapter web
-service URL environment variable, described previously.
+This microservice contains the following projects:
```
Comment (dluc) [inline README.md]: in the lines below, pls fix indentation

File: README.md [hunk 11]
```diff
@@ -100,33 +171,43 @@ the IDE, there are several ways to manage environment variables:
...
-
-{TODO}
-
-Feedback
-==========
+# Troubleshooting
 
-Please enter issues, bugs, or suggestions as GitHub Issues here: 
-https://github.com/Azure/asa-manager-dotnet/issues
+It is possible Azure Stream Analytics may crash. If this happens (rules stop evaluating),
```
Comment (dluc) [inline README.md]: ditto: indentation

File: README.md [hunk 12]
```diff
@@ -2,20 +2,25 @@
 [![Issues][issues-badge]][issues-url]
 [![Gitter][gitter-badge]][gitter-url]
 
-#Azure Stream Analytics Manager overview
+# Azure Stream Analytics Manager Overview
```
Comment (dluc) [inline README.md]: "overview" lowercase IMO

File: README.md [hunk 13]
```diff
@@ -2,20 +2,25 @@
...
 
 This service allows the management the Azure Stream Analytics (ASA) jobs,
 including setting the configuration and starting, stopping, and 
 monitoring their statuses.
 
-The service uses the information managed by 
-the [Telemetry Microservice][telemetry-microservice-url]
-to set the stream processing logic, by implementing monitoring
+The ASA job is supported by two 
+[reference data](https://docs.microsoft.com/en-us/azure/stream-analytics/stream-analytics-use-reference-data)
```
Comment (dluc) [inline README.md]: remove "en-us" from links

File: README.md [hunk 14]
```diff
@@ -2,20 +2,25 @@
...
+sets--one that defines rules and one that defines device groups.
```
Comment (dluc) [inline README.md]: the "--" seems in a strange position, is it needed? should there be spaces?

File: README.md [hunk 15]
```diff
@@ -181,9 +187,9 @@ Please follow our [contribution guidelines](docs/CONTRIBUTING.md).  We love PRs
 
 It is possible Azure Stream Analytics may crash. If this happens (rules stop evaluating), 
 restart ASA in the portal with the following steps:
-1. Navigate to your ASA instance on portal.azure.com. If you deployed an entire 
-Remote Monitoring solution, your instance will be in a resource group with the name
-of your solution.
+1. Navigate to your ASA instance on the [Azure Portal](https://portal.azure.com). 
+If you deployed an entire Remote Monitoring solution, your instance will be in
+a resource group with the name of your solution.
```
Comment (dluc) [inline README.md]: indentation

File: README.md (hunk 12 above)
Comment (molly-moen) [reply README.md]: we are using 'Overview' for all readmes

File: README.md (hunk 10 above)
Comment (molly-moen) [reply README.md]: done

File: README.md (hunk 11 above)
Comment (molly-moen) [reply README.md]: done
